api/src/wfl/module/wgs.clj

```diff
@@ -161,36 +144,46 @@
           out-gs (str (all/slashify output) object)]
       (or (exists? out-gs) (processing? in-gs)))))
 
-(defn start-wgs-workload!
+(defn submit-workload!
   "Use transaction TX to start the WORKLOAD."
-  [tx {:keys [items uuid] :as workload}]
-  (let [env             (get-cromwell-environment workload)
-        default-options (util/make-options env)
-        workload->label {:workload uuid}]
-    (letfn [(submit! [{:keys [id inputs options] :as workflow}]
-              (if (skip-workflow? env workload workflow)
-                [id "skipped" util/uuid-nil]
-                [id "Submitted"
-                 (really-submit-one-workflow
-                  env
-                  inputs
-                  (util/deep-merge default-options options)
-                  workload->label)]))
-            (update! [tx [id status uuid]]
-              (jdbc/update! tx items
-                            {:status status :updated (OffsetDateTime/now) :uuid uuid}
-                            ["id = ?" id]))]
-      (let [now (OffsetDateTime/now)]
-        (run! (comp (partial update! tx) submit!) (:workflows workload))
-        (jdbc/update! tx :workload {:started now} ["uuid = ?" uuid])))))
+  [{:keys [uuid] :as workload}]
+  (let [path            (wdl/hack-unpack-resources-hack (:top workflow-wdl))
+        env             (get-cromwell-environment workload)
+        default-options (util/make-options env)]
+    (letfn [(update-workflow [workflow cromwell-uuid]
+              (assoc workflow :uuid cromwell-uuid
+                     :status "Submitted"
+                     :updated (OffsetDateTime/now)))
+            (submit-batch! [[options workflows]]
+              (map update-workflow
+                   workflows
+                   (cromwell/submit-workflows
+                    env
+                    (io/file (:dir path) (path ".wdl"))
+                    (io/file (:dir path) (path ".zip"))
+                    (map (partial make-cromwell-inputs env) workflows)
+                    (util/deep-merge default-options options)
+                    (merge cromwell-labels {:workload uuid}))))]
+      (let [workflows (group-by #(skip-workflow? env workload %) (:workflows workload))
```

Comment:
`(let [{skipped? true workflows false} (group-by #(skip-workflow? env workload %) (:workflows workload)) …`
or maybe just handle the skipped workflows conditionally in line?